### DISCUSSION TITLE
pag播放过程卡主报错

### DISCUSSION THREAD

**Discussion by DaYang816** [Tencent/libpag#1622] [Q&A] [answered]
Web端，使用的是vue，插件版本"libpag": "^4.2.84"。
一开始是按钮点击主动播放的，然后在播放过程中会出现Uncaught (in promise) DOMException: play() can only be initiated by a user gesture.报错，就卡住了。查了一下跟pag的文件有关系，有些文件正常，有些文件是会卡主，然后我怀疑是不是制作的pag版本和使用插件的pag版本没对上，或者是制作的pag加了什么动效导致触发的。

**Comment by domchen** [comment, marked as answer]
提示已经比较明确了，你的PAG文件里如果包含了BMP预合成的内容就会存在视频播放的需求，视频播放在Web上有些平台是会要求有用户交互才允许的。3个解决方案：1.PAG文件里制作的时候避免使用BMP预合成能力 2.你如果是App带的WebView，是可以在WebView上找下属性去掉这个限制的。3.用外挂的软件解码ffavc来播放视频序列帧，但是性能比较差只能用作兜底：https://pag.art/docs/web-features.html。

**Reply by DaYang816** [reply]
1.我已经给用户播放按钮的交互了，但是它会在播放中途直接卡住，是中途！中途！中途！，卡住后无论主动还是代码再去调用play()是无效的，这个页面就直接卡死了。
2.目前测试下来，安卓的微信浏览器存在这个问题的，UC是根本无法播放（不使用ffavc），华为，小米等自带的浏览都能完整播放结束
3.制作pag插件版本问题，同一个特效，同一文件，再5月份的时候导出应用到代码中播放没问题，7月份重新导出后就会出现中途直接卡住的情况。
4.基于市面上的主流技术框架，并没有给出ffavc的demo，只找到原生写法，所需的插件版本未知
上述问题都是做H5移动端中触发，产品的需求环境也是H5移动端
